docs/configurability-of-serverless.md

```diff
@@ -0,0 +1,76 @@
+# Overview
+
+Following the motivation of [kyma modularisation](https://github.com/kyma-project/community/tree/main/concepts/modularization), serverless should become a module that can be enabled/disabled by the user. 
+Serverless should have its own operator (`serverless-manager`) that should be installed in the target kyma runtime by an infrastructure operators based on the module descriptor (a.k.a `Module Template`).
+`Serverless-manager` watches `Serverless` Custom Resource to re-configure (reconcile) serverless instalation.
+
+## Serverless CR
+
+Reconciliation of serverless components is driven by the content of the `Serverless` Custom Resource.
+
+Serverless CR is a configuration for serverless.
+
+It provides API for configuration capabilities:
+ - user can override tracing backend ( by default jaeger is used if it is discovered )
+ - user can override metrics backend ( by default prometheus is used if it is discovered )
+ - user can enable/disable internal docker registry
+ - user can configure external docker registry configuration
+ - user can override default [build-time container resources](https://kyma-project.io/docs/kyma/latest/05-technical-reference/svls-09-available-presets/)
+ - user can override default [function runtime container resources](https://kyma-project.io/docs/kyma/latest/05-technical-reference/svls-09-available-presets/)
+ - user can configure max simultaneous build jobs
+ - user can set maximum body size accepted by functions
+ - user can modify Log level
+ - ...
+
+ Serverless CR shows the status of serverless module, i.e:
+ - health of serverless workloads (i.e controller, webhook, docker registry installed)
+ - url of the detected event publisher proxy
+ - url of the detected/configured otlp endpoints
+ - indication whether internal docker registry is used / url of configured docker registry
+
+```yaml
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Serverless
+  name: serverless-sample
+spec:
+  dockerRegistry:
+    enableInternal: true
+    externalDockerRegistrySecretName: xxxx 
+  eventingPublisherProxy: http://eventing-publisher-proxy.kyma-system.svc.cluster.local/publish
+  otlpTracesEndpoint: http://tracing-jaeger-collector.kyma-system.svc.cluster.local:2342/v1/metrics ##<-- this is a dummy example
+  otlpMetricsEndpoint: http://tracing-jaeger-collector.kyma-system.svc.cluster.local:4318/v1/trace
+  defaultFunctionRuntimePreset: M
+  defaultFunctionBuildPreset: S
+  maxParallelFunctionBuilds: 5
+  controllerLogLevel: debug
+  ## runtime config
+  maxRequestPayloadSize: 2 #MB
```

Comment:
```suggestion
  maxRequestPayloadSize: 2MB
```
It's better to write it explicit.

Reply:
I'm not sure it will make our life easy with parsing.
How about:
` maxRequestPayloadSizeMB: 2`

Reply:
Do we want to use the same semantics like k8s or we want to allow to set MB only?

Reply:
no .. it make sense to allow flexibility. I guess parsing will not be that hard if we use the same semantics from k8s.
I will adjust the proposal